# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

NAME = "Isa Clevis, #2011221#"
COLLABORATORS = ""

In [7]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [8]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3        V4        V5        V6        V7       
1 0.4686788 0.18458018 0.6467403 0.3977243 0.5002139 0.4476719 0.2833316
2 0.9152612 0.04312403 0.6908749 0.3826101 0.4048803 0.1627602 0.7698906
  V8        V9        V10       V11       V12       V13       V14      
1 0.2771584 0.9052170 0.2975941 0.4312751 0.8396772 0.7192709 0.9180003
2 0.2658353 0.8747347 0.6582490 0.2152192 0.2618015 0.2913037 0.1946896
  V15       V16       V17       V18       V19       V20      
1 0.9229880 0.3444626 0.2264754 0.6307754 0.7046286 0.3278413
2 0.9984835 0.7732869 0.4602841 0.1746964 0.5068655 0.7216031

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [9]:
df2 <- df1 %>%
    gather(Key, Value, V1:V20)
  

head(df2)

Key Value     
1 V1  0.46867878
2 V1  0.91526125
3 V2  0.18458018
4 V2  0.04312403
5 V3  0.64674027
6 V3  0.69087489

## Assignment 2

We will generate some data:

In [10]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [11]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [12]:
df5 <- full_join(df3, df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook